app/src/ui/stashing/stash-diff-header.tsx

```diff
@@ -117,8 +92,16 @@ export class StashDiffHeader extends React.Component<
   private onRestoreClick = async () => {
     const { dispatcher, repository, stashEntry } = this.props
 
-    this.setState({ isRestoring: true }, () => {
-      dispatcher.popStash(repository, stashEntry)
-    })
+    try {
+      this.setState({ isRestoring: true })
+      await dispatcher.popStash(repository, stashEntry)
+    } catch (err) {
+      dispatcher.showPopup({
+        type: PopupType.Error,
+        error: err,
+      })
```

Comment:
I _think_ this should be more correct:
```suggestion
      dispatcher.postError(err)
```

Reply:
Thanks for the suggestion.
It is committed.

Reply:
> I _think_ this should be more correct:
> 
> ```suggestion
> 
>       dispatcher.postError(err)
> 
> ```